jupyterlab/labapp.py

```diff
@@ -258,6 +260,91 @@ def start(self):
         print('Workspaces directory: %s' % get_workspaces_dir())
 
 
+class LabWorkspaceListApp(JupyterApp):
+    version = version
+    description = """
+    Print all the workspaces available
+
+    If '--json' flag is passed in, 'json' output of each workspace is printed.
+    If '--jsonlist' flag is passed in, 'json' output of list of workspaces is printed.
+    If nothing is passed in, 'human-readable' output is printed.
+    """
+    flags = dict(
+        jsonlist=(
+            {"LabWorkspaceListApp": {"jsonlist": True}},
+            _i18n("Produce machine-readable JSON list output."),
+        ),
+        json=(
+            {"LabWorkspaceListApp": {"json": True}},
+            _i18n("Produce machine-readable JSON object on each line of output."),
+        ),
+    )
+
+    jsonlist = Bool(
```

Comment:
perhaps `jsonl` or `jsonlines`, and reference https://jsonlines.org so the format is _slightly_ more specified